File: src/test/java/com/YuQ/test/controller/TestController.java
package com.YuQ.test.controller;

import com.IceCreamQAQ.YuQ.annotation.Action;
import com.IceCreamQAQ.YuQ.annotation.Before;
import com.IceCreamQAQ.YuQ.annotation.GroupController;
import com.IceCreamQAQ.YuQ.annotation.Inject;

@GroupController
public class TestController {

    @Before
    public void before(Long qq,@Inject(name = "group") Long group){

    }

    @Action(value = "菜单",intercept = true)
    public String menu(){
//        throw new RuntimeException();
        return "Group OK!";
    }

}

File: src/test/kotlin/com/icecreamqaq/yu/test/controller/TestController.kt
package com.icecreamqaq.yu.test.controller

import com.IceCreamQAQ.Yu.annotation.Action
import com.IceCreamQAQ.Yu.annotation.Before
import com.IceCreamQAQ.Yu.annotation.DefaultController
import com.IceCreamQAQ.Yu.annotation.PathVar
import com.icecreamqaq.yu.test.annotation.CController
import javax.inject.Named

@DefaultController
class TestController {

    @Before
    fun testBefore():String{
        return "Test Before"
    }

    @Action("test")
    fun testAction(@PathVar(value = 2, type = PathVar.Type.number) aaa: String, bbb: String, @Named("111") ccc: String) {
        println("before = $aaa")
        println("Test Action")
    }

}